### PR TITLE
update locations method

### PR DESCRIPTION
**Pull request by duck7000**
- Rewrite to new xpath function.
- Add movielocation because it is missing. Backwards compatible as it adds a array item.

### CODE REVIEW

**Review by tboothman — CHANGES_REQUESTED** [review]
I'm not sure I want to merge this as it's not backwards compatible.
The signature of the method is different now. It was an array of strings before now it's an array of arrays.
The tests on this method fail for that reason.

The two field names are ambiguous. The imdb editing page calls them location and attribute - it would make sense to stick with their naming
image

The results were constrained to the `section[@id=\"filming_locations\"]` section before which is good practice for parsing a changing website as you never know when they might add in a new table.

**Comment by duck7000** [comment]
Thanks for your comments.

backwards compatible can't be achieved with 2 values, so atribute has to go then?
Or i have to combine those 2 values in one string, which is also not backwards compatible.

I agree with the field names, although attribute doesn't say much what it stands for

**Comment by duck7000** [comment]
I adjusted it back to what it was, so keep backwards compatible.
I don't see a way to include attribute but keep backwards compatibility?

**Comment by jreklund** [comment]
The only way to not break backwards compatibility are to add params, to the method, to change the return type/behavior.
In case that's not something you want to develop, you also need to revert the phpdoc block specifying `@return`.

**Comment by duck7000** [comment]
@jreklund 
Thanks for your feedback.

For now i leave it like it is, i have updated phpdoc.

File: src/Imdb/Title.php [hunk 1]
```diff
@@ -2614,15 +2614,13 @@ public function releaseInfo()
...
+            if (empty($xpath)) {
                 return array();
             } // no such page
-            $doc = new \DOMDocument();
-            @$doc->loadHTML($page);
-            $xp = new \DOMXPath($doc);
-            $cells = $xp->query("//section[@id=\"filming_locations\"]//dt");
+            $cells = $xpath->query("//section[@id=\"filming_locations\"]//dt");
             foreach ($cells as $cell) {
+                $dt = $xpath->query($cell->getNodePath() . '//dt')->item(0)->nodeValue;
```
Comment (jreklund) [inline src/Imdb/Title.php:2623]: It seems travis are crashing on this one. Can't see it beeing used, so should revert that one as well I'm afraid.

**Review by jreklund — CHANGES_REQUESTED** [review]

File: src/Imdb/Title.php (hunk 1 above)
Comment (duck7000) [reply src/Imdb/Title.php:2623]: ...oeps left over i guess

**Review by jreklund — APPROVED** [review]

**Comment by duck7000** [comment]
@all
Thanks for merging!

**Pull request merged**